lib/chef/provider/package/apt.rb

```diff
@@ -216,7 +218,9 @@ def resolve_package_versions(pkg)
         end
 
         def resolve_virtual_package_name(pkg)
-          showpkg = run_noninteractive("apt-cache", "showpkg", pkg).stdout
+          # apt-cache considers package names as regex by default. The anchor flag will decide whether to match name exact string or not
+          resolved_pkg = new_resource.anchor_package_name_patterns ? pkg : "^#{pkg}$"
```

Comment:
rather than repeat this line/logic over and over, please extract this to a method (and I"m reversing the log here per my previous comment):

```ruby
def resolved_package(pkg)
  new_resource.anchor_package_name_patterns ? "^#{pkg}$" : pkg
end
```